presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/thrift/ThriftHiveMetastoreClient.java

```diff
@@ -510,30 +561,60 @@ public void dropConstraint(String dbName, String tableName, String constraintNam
             throws TException
     {
         DropConstraintRequest dropConstraintRequest = new DropConstraintRequest(dbName, tableName, constraintName);
+        if (catalogName.isPresent()) {
+            dropConstraintRequest.setCatName(catalogName.get());
+        }
         client.drop_constraint(dropConstraintRequest);
     }
 
     @Override
     public void addUniqueConstraint(List<SQLUniqueConstraint> constraint)
             throws TException
     {
-        AddUniqueConstraintRequest addUniqueConstraintRequest = new AddUniqueConstraintRequest(constraint);
+        List<SQLUniqueConstraint> updatedConstraints = constraint;
+        if (catalogName.isPresent()) {
+            updatedConstraints = constraint.stream().map(c -> {
```

Comment:
nit: we tend to use full names for lambda parameters:
```suggestion
            updatedConstraints = constraint.stream().map(uniqueConstraint -> {
```